src/main/scala/firrtl/passes/RemoveCHIRRTL.scala

```diff
@@ -82,12 +82,10 @@ object RemoveCHIRRTL extends Transform {
       def set_enable(vec: Seq[MPort], en: String) = vec map (r =>
         Connect(sx.info, SubField(SubField(Reference(sx.name, ut), r.name, ut), en, BoolType), zero)
       )
-      def set_write(vec: Seq[MPort], data: String, mask: String) = vec flatMap {r =>
+      def set_write(vec: Seq[MPort], data: String, mask: String) = vec flatMap { r =>
         val tmask = createMask(sx.tpe)
-        IsInvalid(sx.info, SubField(SubField(Reference(sx.name, ut), r.name, ut), data, tdata)) +:
-             (create_exps(SubField(SubField(Reference(sx.name, ut), r.name, ut), mask, tmask))
-               map (Connect(sx.info, _, zero))
-             )
+        val portRef = SubField(Reference(sx.name, ut), r.name, ut)
+        Seq(IsInvalid(sx.info, SubField(portRef, data, tdata)), IsInvalid(sx.info, SubField(portRef, mask, tmask)))
```

Comment:
To solidify my understanding, you are invalidating the mask instead of zeroing it because the write enable takes care of any when statements between the memory and write port declaration?

Reply:
Exactly, the write enable signal is inferred as the conjunction of when conditions between memory declaration and port declaration. In order to not duplicate that condition on the mask, this PR is inferring the mask as the conjunction of when conditions between declaration of the port and the actual connection to the subfield or subindex of the write port.